eng/servicingValidation.targets

```
@@ -0,0 +1,35 @@
+<Project>
+
+  <!-- Flows the list of ProjectReferences that are enabled for packaging when building a multi-targeting project -->
+  <Target Name="_AddTransitiveServicedPackagesToOutput"
+          AfterTargets="GetTargetPathWithTargetPlatformMoniker"
+          Condition="'$(IsInnerBuild)' == 'true'">
+    <PropertyGroup>
+      <_TransitiveServicedPackages
+          Condition="'%(ReferencePath.ReferenceSourceTarget)' == 'ProjectReference' and
+                     '%(ReferencePath.IsPackable)' == 'true'"
+                     >@(ReferencePath->'%(OriginalProjectReferenceItemSpec)')</_TransitiveServicedPackages>
+    </PropertyGroup>
+    <ItemGroup>
+      <TargetPathWithTargetPlatformMoniker TransitiveServicedPackages="$(_TransitiveServicedPackages)" />
+    </ItemGroup>

```

I would leverage the msbuild item functions here to make this more readable and avoid putting an item into a property.

```suggestion
    <ItemGroup>
      <TargetPathWithTargetPlatformMoniker TransitiveServicedPackages="@(ReferencePath->WithMetadataValue('ReferenceSourceTarget', 'ProjectReference')->WithMetadataValue('IsPackable', 'true')->Metadata('OriginalProjectReferenceItemSpec'))" />
    </ItemGroup>
```

---

See https://learn.microsoft.com/en-us/visualstudio/msbuild/item-functions?view=vs-2022

`WithMetadataValue`

> Returns items that have the given metadata name and value. The comparison is case insensitive.

`Metadata`

> Returns the values of the metadata that have the metadata name. The items returned have the same metadata as the source values.

Item functions are displayed much nicer in binlogs than with item batching (`%(...)`). I also find them more readable when looking at the code. They are also more performant but that is neglectable.